contracts/distribution/HAMAMPLPool.sol

```diff
@@ -663,7 +676,7 @@ contract HAMAMPLPool is LPTokenWrapper, IRewardDistributionRecipient {
     function lastTimeRewardApplicable() public view returns (uint256) {
         return Math.min(block.timestamp, periodFinish);
     }
-
+    ///@dev below needs complete rework for gas, readability, and distro fxnlty
```

Comment:
I think these "TODO" style comments might be better placed into a separate file. Generally would expect these comments to just be code comments. It will be quite difficult to track these all down. Maybe at least prefix with `TODO: ` so they are searchable?